ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -481,6 +569,39 @@ private void updateContainersWithMySqlStore(Account account, Collection<Containe
     return new Pair<>(addedContainers, updatedContainers);
   }
 
+  /**
+   * Updates {@link Account}s to in-memory cache.
+   * @param accounts added or modified {@link Account}s
+   */
+  private void updateAccountsInCache(Collection<Account> accounts) {
+    infoMapLock.writeLock().lock();
+    try {
+      accountInfoMapRef.get().addOrUpdateAccounts(accounts);
+    } finally {
+      infoMapLock.writeLock().unlock();
+    }
+  }
+
+  /**
+   * Updates {@link Container}s to in-memory cache.
+   * @param containers added or modified {@link Container}s
+   */
+  private void updateContainersInCache(Collection<Container> containers) {
+
+    infoMapLock.writeLock().lock();
+    try {
+      accountInfoMapRef.get().addOrUpdateContainers(containers);
+    } finally {
+      infoMapLock.writeLock().unlock();
+    }
+
+    // Remove containers from not-found LRU cache.
+    for (Container container : containers) {
+      recentNotFoundContainersCache.remove(
```

Comment:
This negates my earlier comment in response to Yingyi's question.

Reply:
I guess I was trying to say in the case I mentioned if container name is in `recentNotFoundContainersCache` it requires periodic refresh to update cache (remove this container name out of "NotFound" cache). On-demand fetch in `getContainerByName` won't be performed. 
Anyway, I think it's fine to rely on periodic refresh (and probably it's the only way we can do).